compiler/circle-mpqsolver/src/core/DumpingHooks.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __MPQSOLVER_DUMPING_HOOKS_H__
+#define __MPQSOLVER_DUMPING_HOOKS_H__
+
+#include <luci/IR/Module.h>
+
+#include <core/Quantizer.h>
+#include <core/SolverHooks.h>
+#include <core/Dumper.h>
+
+#include <string>
+
+namespace mpqsolver
+{
+namespace core
+{
+
+class DumpingHooks : public QuantizerHook, public SolverHooks
+{
+public:
+  DumpingHooks();
+
+  /**
+   * @brief whether all intermediate artifacts whould be saved
+   */
+  void set_save_intermediate(bool save, const std::string &save_path);
+
+  virtual void on_quantized_ready(luci::Module *module) const override;
+
+  virtual void on_new_run(const std::string &model_path) override;
+  virtual void on_begin_iterations() override;
+  virtual void on_end_iterations() override;
+  virtual void on_new_iteration() override;
+
+  virtual void on_Q8_error(float error) const override;
+  virtual void on_Q16_error(float error) const override;
+  virtual void on_MPQ_error(float error) const override;
+  virtual void on_MPQ_configuration(const LayerParams &layers,
+                                    const std::string &def_dtype) const override;
+
+  virtual void on_final_MPQ(const LayerParams &layers, const std::string &def_dtype) const override;
```

Comment:
Hmm.. The hook names are a bit confusing to me. And I think they can be much simplified.

on_quantized_ready -> on_quantized

on_new_run, on_begin_iterations, on_Q8_error, on_Q16_error -> on_begin_solver
on_final_MPQ, on_end_iterations -> on_end_solver
on_new_iteration -> on_begin_iteration
on_MPQ_configuration, on_MPQ_error -> on_end_iteration

Reply:
Ahh. Thank you. Ok.

Reply:
@jinevening 
done.